README.md

```diff
@@ -1,35 +1,133 @@
+git clone https://github.com/cfra/heroku-buildpack-oauth2-proxy -b example-static oauth2-example
+cd oauth2-example
+```
+
+Then, create a new Heroku application and set the buildpacks:
+
+```console
+heroku create
+heroku buildpacks:set https://github.com/heroku/heroku-buildpack-static.git
 heroku buildpacks:add cfra/oauth2-proxy
 ```
 
-## Starting on heroku
+You need an account for the proxy so that it can interface with your OAuth provider.
+For this example, we will be using Github.
 
-You need to ensure both your backend and `oauth2_proxy` are started by your
-`Procfile`. The easiest way is the provided script `start_with_oauth2_proxy.sh`.
+Create a new OAuth app in the [Github developer settings](https://github.com/settings/developers).
+Ensure that the "Authorization callback URL" is set correctly. For example, if you are using
+`herokuapp.com` and your application is called `ancient-woodland-33672`, you should set the
+callback URL to: `https://ancient-woodland-33672.herokuapp.com/oauth2/callback`.
 
-Pass in the call to start your backend as its arguments. For example, the
-[heroku-buildpack-static](https://github.com/heroku/heroku-buildpack-static)
-has `/app/bin/boot` as entrypoint. Given that, a usable `Procfile` to run it
-with the proxy could look as following:
+After having created the OAuth app successfully, you will be shown a _Client ID_ and a _Client Secret_,
+which you need to configure on the Heroku app:
+
+```console
+heroku config:set OAUTH2_PROXY_CLIENT_ID=0123456789abcdef1234
+heroku config:set OAUTH2_PROXY_CLIENT_SECRET=0123456789abcdef0123456789abcdef01234567
+```
+
+Furthermore, you need to specify that Github should be used as authentication provider
+and provide a secret key for encrypting the session cookies:
+
+```console
+heroku config:set OAUTH2_PROXY_PROVIDER=github
+heroku config:set OAUTH2_PROXY_COOKIE_SECRET=$(python -c \
+    'from secrets import token_urlsafe; print(token_urlsafe(32)[:32])' \
+)
+```
+
+Now, you are all set and can start the first deployment of your example application:
+
+```console
+git push heroku HEAD:master
+```
+
+You will see the buildlog for your application, followed by a message about successful
+deployment.
+
+When you now navigate to your application on Heroku (following the name from the example it would
+be at `https://ancient-woodland-33672.herokuapp.com/`) you will be prompted to login with Github.
+
+After successful authentication, Github will redirect you back to your application and a success
+page will be shown.
+
+For futher steps, you might want to have a look in the [configuration Section](#configuration) to
+learn about configuration options for finer grained authentication, e.g. allowing only members
+of a particular Github organization.
+
+## Usage with other applications
+
+For Using this buildpack with your application, you need to do two things:
+
+On the one hand, you need to set up the configuration for `oauth2_proxy`. The getting started section
```

Comment:
You could also use a numerated list. Maybe this is easier to read?

Reply:
I am not completely sure what you have in mind here.

My intention was to describe the general considerations necessary for using this buildpack in this section, not to provide a step by step guide on how to use it with any application, because this is probably hard to do anyway.

Could you maybe provide a more detailed example? :pray:

Reply:
The sentence above says "you need to do two things". So I thought I might be easier to understand if this is a numerated list:

```markdown
1. You need to set up the configuration for `oauth2_proxy`. The getting started section describes
   this process for GitHub. For other providers, a look at the [configuration Section](#configuration)
   and at the [OAuth2 Provider Configuration documentation of oauth2\_proxy](https://github.com/pusher/oauth2_proxy#oauth-provider-configuration)
   will provide you with the necessary information.
2. You need to ensure that `oauth2_proxy` is run as a reverse proxy in front of your actual
   application worker.
```

Reply:
Since those two things are not small steps, but rather long descriptions of concepts, an enumeration doesn't seem to work very well here. I tried it and in my humble opinion it makes the document less readable by adding an unnecessary indentation and provides little value with respect to the structure.